test/javascripts/acceptance/chat-quoting-test.js

```diff
@@ -0,0 +1,235 @@
+import { test } from "qunit";
+import { click, currentURL, visit } from "@ember/test-helpers";
+import {
+  acceptance,
+  exists,
+  query,
+  visible,
+} from "discourse/tests/helpers/qunit-helpers";
+import {
+  chatChannels,
+  chatView,
+} from "discourse/plugins/discourse-chat/chat-fixtures";
+import selectKit from "discourse/tests/helpers/select-kit-helper";
+
+let quoteResponse = {
+  bbcode: `[chat quote="martinchat;3875498;2022-02-04T01:12:15Z" channel="The Beam Discussions"]
+  an extremely insightful response :)
+  [/chat]`,
+};
+
+function setupPretenders(server, helper) {
+  server.get("/chat/chat_channels.json", () => helper.response(chatChannels));
+  server.post(`/chat/9/quote.json`, () => helper.response(quoteResponse));
+  server.post(`/chat/4/quote.json`, () => helper.response(quoteResponse));
+  server.post(`/chat/7/quote.json`, () => helper.response(quoteResponse));
+  server.get("/chat/:chatChannelId/messages.json", () =>
+    helper.response(chatView)
+  );
+  server.post("/uploads/lookup-urls", () => {
+    return helper.response([]);
+  });
+}
+
+acceptance("Discourse Chat | quoting out of topic", function (needs) {
+  needs.user({
+    admin: false,
+    moderator: false,
+    username: "eviltrout",
+    id: 1,
+    can_chat: true,
+    has_chat_enabled: true,
+  });
+
+  needs.settings({
+    chat_enabled: true,
+  });
+
+  needs.pretender((server, helper) => {
+    setupPretenders(server, helper);
+  });
+
+  test("it opens the composer and appends the quote", async function (assert) {
+    await visit("/chat/channel/9/Site");
+    const firstMessage = query(".chat-message-container");
+    const dropdown = selectKit(".chat-message-container .more-buttons");
+    await dropdown.expand();
+    await dropdown.selectRowByValue("selectMessage");
+
+    assert.ok(firstMessage.classList.contains("selecting-messages"));
+    const quoteBtn = query(".chat-live-pane #chat-quote-btn");
+    assert.equal(
+      quoteBtn.disabled,
+      false,
+      "button is enabled as a message is selected"
+    );
+    await click("#chat-quote-btn");
+    assert.ok(exists("#reply-control.composer-action-createTopic"));
```

Comment:
not related to your changes but the casing on this class name is so weird.

Reply:
Yeah it's because it comes directly from the composer actions defined here:

https://github.com/discourse/discourse/blob/febc2361bd99be26317d9fe06c8024b2696e887a/app/assets/javascripts/discourse/app/models/composer.js#L37-L44